pyat/at/tracking/particles.py

```diff
@@ -0,0 +1,131 @@
+                            blength] 
+                            computes the 6x6 sigma matrix.
+
+    KEYWORDS
+        twiss=False     Flag which states whether the input is a
+                        lattice object or list of twiss parameters
+
+    OUTPUT
+        sigma_matrix    correlation matrix (either 2x2, 4x4 or 6x6)
+
+    """
+    if not twiss:
+        cflag = False
+        if not argin.radiation:
+            argin.radiation_on()
+            cflag = True
+
+        emit0, beamdata, emit = ohmi_envelope(argin, refpts=[0])
+        sig_matrix = emit.r66[0]
+
+        if cflag:
+            argin.radiation_off()
+
+
+    else:
+        if len(argin) == 2:
+            [espread, blength] = argin
+            sig_matrix = numpy.array([
+                            [espread*espread, 0],
+                            [0, blength*blength]
+                                     ])
+
+        elif len(argin) == 3:
+            [bx, ax, epsx] = argin
+            sig_matrix = epsx * numpy.array([
+                                    [bx, -ax],
+                                    [-ax, (1 + ax * ax)/bx]
+                                            ])
+
+        elif len(argin) == 6:
+            [bx, ax, epsx, by, ay, epsy] = argin
+            sig_matrix = numpy.bmat([
+                            [sigma_matrix([bx, ax, epsx], twiss=True),
+                                numpy.zeros((2, 2))],
+                            [numpy.zeros((2, 2)),
+                                sigma_matrix([by, ay, epsy], twiss=True)]
+                                    ])
+
+        elif len(argin) == 8:
+            [bx, ax, epsx, by, ay, epsy, espread, blength] = argin
+            sig_matrix = numpy.bmat([
+                            [sigma_matrix([bx, ax, epsx], twiss=True),
+                                numpy.zeros((2, 4))],
+                            [numpy.zeros((2, 2)),
+                                sigma_matrix([by, ay, epsy], twiss=True),
+                                numpy.zeros((2, 2))],
+                            [numpy.zeros((2, 4)),
+                                sigma_matrix([espread, blength], twiss=True)]
+                                    ])
+
+    return sig_matrix
+
+
+def beam(np, sigma, orbit=None):
+    """
+    Generates an array of random particles according to the given sigma
+    matrix
+
+    PARAMETERS
+        np              Number of particles
+        sigma           sigma_matrix as calculated by at.sigma_matrix
+
+    KEYWORDS
+        orbit=None      An orbit can be provided to give a center of 
+                        mass offset to the distribution
+
+    OUTPUT
+        particle_dist   a matrix of shape (M, np) where M is shape of
+                        sigma matrix
+    """
+    v = numpy.random.normal(size=(sigma.shape[0], np))
+
+    try:
+        lmat = numpy.linalg.cholesky(sigma).T
+#        needs the .T to be the same as matlab
```

Comment:
Since later you use
`numpy.dot(lmat.T, v)`
you can avoid transposing twice!

Reply:
you're right. Fixed.